libr/anal/p/anal_sbpf.c

```diff
@@ -487,11 +498,34 @@ static bool sbpf_analyze_strings(RAnal *anal) {
 
 				// Create flag with format: ptr.<pointer_addr>_<string>
 				snprintf (flagname, sizeof (flagname), "ptr.%"PFMT64x"_%s", ref->addr, safe_str);
+
+				// Save string for comment (unfiltered for readability)
+				ut32 comment_len = size < 250 ? size : 250;
+				r_str_ncpy (comment_str, (char *)str_buf, comment_len + 1);
+
+				// Replace non-printable chars with dots for the comment
+				ut32 i;
+				for (i = 0; i < comment_len; i++) {
+					if (comment_str[i] < 0x20 || comment_str[i] > 0x7e) {
```

Comment:
we have `IS_PRINTABLE` macro for that. which is used by the hexdump and other strings functions. also maybe you can just filter that thing while copying it instead of iterating it twice ( r_str_ncpy + for )